cpp/include/Ice/Service.h

```diff
@@ -91,120 +67,88 @@ namespace Ice
         [[nodiscard]] Ice::CommunicatorPtr communicator() const;
 
-        /// Obtains the Service singleton.
+        /// Gets the Service singleton.
         /// @return A pointer to this service.
         static Service* instance();
 
         /// Indicates whether the program is running as a Win32 service or Unix daemon.
         /// @return `true` if the program is running as a service, `false` otherwise.
         [[nodiscard]] bool service() const;
 
-        /// Obtains the program name. If the program is running as a Win32
-        /// service, the return value is the service name. Otherwise the
-        /// return value is the executable name (i.e., argv[0]).
+        /// Gets the program name. If the program is running as a Win32 service, the return value is the service name.
+        /// Otherwise the return value is the executable name (i.e., argv[0]).
         /// @return The service name.
         [[nodiscard]] std::string name() const;
 
-#ifdef _WIN32
-        /// Alternative entry point for services that use their own
-        /// command-line options. Instead of invoking main(), the
-        /// program processes its command-line options and invokes
-        /// run(). To run as a Win32 service or Unix daemon, the
-        /// program must first invoke configureService() or
-        /// configureDaemon(), respectively.
-        ///
-        /// @param argc Specifies the number of arguments in argv.
-        /// @param argv The command-line arguments.
-        /// @param initData Configuration data for the new Communicator.
-        /// @return The application's exit status: EXIT_FAILURE or EXIT_SUCCESS.
-        int run(int argc, const wchar_t* const argv[], InitializationData initData = {});
-#endif
-
-        /// Alternative entry point for services that use their own
-        /// command-line options. Instead of invoking main(), the
-        /// program processes its command-line options and invokes
-        /// run(). To run as a Win32 service or Unix daemon, the
-        /// program must first invoke configureService() or
-        /// configureDaemon(), respectively.
-        ///
-        /// @param argc Specifies the number of arguments in argv.
+        /// Alternative entry point for services that use their own command-line options. Instead of calling #main, the
+        /// program processes its command-line options and calls #run. To run as a Win32 service or Unix daemon, the
+        /// program must first call `configureService` or #configureDaemon, respectively.
+        /// @param argc The number of arguments in @p argv.
         /// @param argv The command-line arguments.
-        /// @param initData Configuration data for the new Communicator.
+        /// @param initData Configuration data for the new communicator.
         /// @return The application's exit status: EXIT_FAILURE or EXIT_SUCCESS.
         int run(int argc, const char* const argv[], InitializationData initData = {});
 
 #ifdef _WIN32
```

Comment:
This `#ifdef _WIN32` doc doesn't work, obviously. 

We could define _WIN32 when building the doc, but then we'd lose the `#else`. A solution would be to split this class in Ice::WindowsService and Ice::DaemonService, and document both ... but quite a bit of work for this rarely used class.

Maybe we should deprecate this class for Ice 3.8?

Reply:
For `Ice/SSL/ClientAuthenticationOptions.h`

I use 

```
#if defined(ICE_USE_SCHANNEL)
endif

#if defined(ICE_USE_SECURE_TRANSPORT)
endif
```

This way doxygen would seem both, but maybe not worth here.